ampd/Dockerfile

```diff
@@ -1,10 +1,32 @@
-FROM rust:bookworm as build
-WORKDIR /app
+FROM rust:1.74-bookworm as builder
 RUN apt-get update && apt-get install -y clang protobuf-compiler
-COPY . .
-RUN cargo build --release
 
-FROM debian:bookworm-slim AS final
-COPY --from=build /app/target/release/ampd /usr/local/bin/ampd
-WORKDIR /usr/local/bin
-ENTRYPOINT ["ampd"]
+WORKDIR /ampd
+COPY ./Cargo.toml .
+COPY ./ampd/Cargo.toml ./ampd/Cargo.toml
+COPY ./ampd/Cargo.lock ./ampd/Cargo.lock
+COPY ./packages ./packages
+COPY ./contracts ./contracts
+COPY ./integration-tests ./integration-tests
+COPY ./.cargo ./.cargo
+
+# build dependencies separately
+RUN mkdir ampd/src && echo 'fn main() {}' > ampd/src/main.rs
+RUN --mount=type=ssh cargo build --release 
```

Comment:
`--mount=type=ssh`
 This is a public github repo, do we still need this?

Reply:
I don't think its needed in this case. Gotta double check.